uefi-mkconfig

```diff
@@ -208,18 +211,19 @@ main () {
 	kernel_commands="$strip_kernel_commands"
 
         [[ -z $valid_kernel_commands ]] && ewarn "Warning, kernel command \"root=\" is missing from loaded configuration!"
-		
+
 	# Clear old entries for regeneration
-	## 256..512 is because entry IDs are actually in hexadecimal format
-	for entry_number in {256..512}; do
-		# Wipe only boot entries with UMC stamp
-		if [[ "$(efibootmgr -u | grep Boot$(printf %04X $entry_number))" == *"UMC "* ]]; then
-			efibootmgr -q -B -b $(printf %04X $entry_number)
-		fi
+	## 256..512 (so 0100-0200 because entry IDs are actually in hexadecimal format)
+	for entry_number in $(efibootmgr -u|grep -Po '[Bb]oot0(1[0-9a-fA-F]{2}|200)\W{1,2}UMC'|grep -Po '[0-9a-fA-F]{4}'); do
+	        efibootmgr -q -B -b $entry_number
 	done
 
 	for partition in $mounted_efi_partitions; do
-		
+
+                kernel_images_old=()
+                kernel_images_new=()
+                partition_efis_main=()
```

Comment:
Let's also make these `local`, we don't use these in the other functions as far as I can see